### PR TITLE
nautilus: pybind/mgr/volumes: deadlock on async job hangs finisher thread

### PR DESCRIPTION
**Pull request by batrick**
https://tracker.ceph.com/issues/50128

### CODE REVIEW

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
@ceph/dashboard I don't think the API test failure is caused by this PR. Is nautilus broken? Thanks for the help.

**Comment by batrick** [comment]
(fixing jenkins test failures)

**Review by vshankar — APPROVED** [review]
Couple of commits that have conflicts do not have description on conflict resolution. Is that intentional?

Otherwise, LGTM.

**Comment by vshankar** [comment]
> Couple of commits that have conflicts do not have description on conflict resolution. Is that intentional?
> 
> Otherwise, LGTM.

@batrick is on PTO. I'm OK with the above. Good to go once nautilus runs are verified.

File: src/pybind/mgr/mgr_util.py
Comment (ajarr) [inline src/pybind/mgr/mgr_util.py:34]: This hits an error while testing,
```
2021-05-24T22:31:43.185 INFO:tasks.ceph.mgr.x.smithi039.stderr:2021-05-24 22:31:43.165 7ff0be342700 -1 mgr load Traceback (most recent call last):
2021-05-24T22:31:43.185 INFO:tasks.ceph.mgr.x.smithi039.stderr:  File "/usr/share/ceph/mgr/volumes/module.py", line 338, in __init__
2021-05-24T22:31:43.185 INFO:tasks.ceph.mgr.x.smithi039.stderr:    self.vc = VolumeClient(self)
2021-05-24T22:31:43.186 INFO:tasks.ceph.mgr.x.smithi039.stderr:  File "/usr/share/ceph/mgr/volumes/fs/volume.py", line 58, in __init__
2021-05-24T22:31:43.186 INFO:tasks.ceph.mgr.x.smithi039.stderr:    self.cloner.queue_job(fs['mdsmap']['fs_name'])
2021-05-24T22:31:43.186 INFO:tasks.ceph.mgr.x.smithi039.stderr:  File "/usr/share/ceph/mgr/volumes/fs/async_job.py", line 215, in queue_job
2021-05-24T22:31:43.186 INFO:tasks.ceph.mgr.x.smithi039.stderr:    with lock_timeout_log(self.lock):
2021-05-24T22:31:43.187 INFO:tasks.ceph.mgr.x.smithi039.stderr:  File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
2021-05-24T22:31:43.187 INFO:tasks.ceph.mgr.x.smithi039.stderr:    return self.gen.next()
2021-05-24T22:31:43.187 INFO:tasks.ceph.mgr.x.smithi039.stderr:  File "/usr/share/ceph/mgr/mgr_util.py", line 34, in lock_timeout_log
2021-05-24T22:31:43.187 INFO:tasks.ceph.mgr.x.smithi039.stderr:    locked = lock.acquire(timeout=timeout)
2021-05-24T22:31:43.188 INFO:tasks.ceph.mgr.x.smithi039.stderr:TypeError: acquire() takes no keyword arguments
```

http://qa-proxy.ceph.com/teuthology/yuriw-2021-05-24_19:18:13-fs-wip-yuri3-testing-2021-05-24-0956-nautilus-distro-basic-smithi/6132079/teuthology.log

@vshankar seems like this is not py2 compatible. Not sure why teuthology is testing py2

Comment (vshankar) [reply src/pybind/mgr/mgr_util.py:34]: That's right -- Py2 lock.acquire() does not take `timeout` as arg.

Comment (vshankar) [reply src/pybind/mgr/mgr_util.py:34]: We could probably just drop commit fef88bcc7c9fd866c762ee8d4b9239f2c8c9af36 since its there to catch deadlocks and does not change any functionality as such.

@rraja @fullerdj @batrick

Comment (ajarr) [reply src/pybind/mgr/mgr_util.py]: @vshankar thanks! let's drop it for now.

**Comment by vshankar** [comment]
updated -- dropped commit fef88bc

File: src/pybind/mgr/mgr_util.py
Comment (vshankar) [reply src/pybind/mgr/mgr_util.py]: Just FYI, we haven't run into the suspected deadlock (for global lock) in master and other branches. So its OK to drop the above mentioned commit (which adds debugging for the suspected deadlock).

**Review by ajarr — COMMENTED** [review]
@yuriw please don't merge this PR too. This needs core team approval.

**Review by ideepika — APPROVED** [review]
Core testing and PR looks good to me, thanks!

**Comment by cfsnyder** [comment]
> updated -- dropped commit [fef88bc](https://github.com/ceph/ceph/commit/fef88bcc7c9fd866c762ee8d4b9239f2c8c9af36)

@vshankar  should this commit also be dropped in the Octopus backport?

**Comment by vshankar** [comment]
> @vshankar should this commit also be dropped in the Octopus backport?

Octopus supports Py3. Good to include the above commit.

**Pull request merged**
